homeassistant/components/esphome/assist_satellite.py

```diff
@@ -784,3 +810,64 @@ def send_audio_bytes(self, data: bytes) -> None:
             return
 
         self.transport.sendto(data, self.remote_addr)
+
+
+async def async_setup(hass: HomeAssistant) -> None:
+    """Set up the satellite."""
+    wake_words_dir = Path(hass.config.path(WAKE_WORDS_DIR_NAME))
+    wake_words = hass.data.setdefault(_DATA_WAKE_WORDS, {})
+
+    def _cache_wake_words() -> None:
+        for config_path in wake_words_dir.glob("*.json"):
+            wake_word_id = config_path.stem
+            model_path = config_path.with_suffix(".tflite")
+            if not model_path.exists():
+                # Missing model file
+                continue
+
+            with open(config_path, encoding="utf-8") as config_file:
+                config_dict = json.load(config_file)
+                try:
+                    config = _WAKE_WORD_CONFIG_SCHEMA(config_dict)
+                except vol.Invalid as err:
+                    # Invalid config
+                    _LOGGER.debug(
+                        "Invalid wake word config: path=%s, error=%s",
+                        config_path,
+                        humanize_error(config_dict, err),
+                    )
+                    continue
+
+                with open(model_path, "rb") as model_file:
+                    model_hash = hashlib.sha256(model_file.read()).hexdigest()
+
+                model_size = model_path.stat().st_size
+                config_rel_path = config_path.relative_to(wake_words_dir)
+
+                # Only intended for the internal network
+                base_url = get_url(hass, prefer_external=False, allow_cloud=False)
+
+                wake_words[wake_word_id] = VoiceAssistantExternalWakeWord.from_dict(
+                    {
+                        "id": wake_word_id,
+                        "wake_word": config["wake_word"],
+                        "trained_languages": config_dict.get("trained_languages", []),
+                        "model_type": config["type"],
+                        "model_size": model_size,
+                        "model_hash": model_hash,
+                        "url": f"{base_url}{WAKE_WORDS_API_PATH}/{config_rel_path}",
+                    }
+                )
+
+    # Cache wake word models in config/custom_wake_words
+    await hass.async_add_executor_job(_cache_wake_words)
```

Comment:
So I would delay this call until the first assist device is actually connected. You can use [the singleton helper](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/singleton.py#L35) for that.

And since the singleton helper is based on a hass.data key, popping that would kill the cache.